### ISSUE TITLE
future training: area<=0 check for ann_file parsing, dataset.py will not load bbox only

### ISSUE THREAD

**Issue by lessw2020** [rwightman/efficientdet-pytorch#10]
I'm picking up the code from dataset.py and noticed all my custom bounding boxes were coming up empty. 
This is b/c of the check on 71 in dataset.py:
`            if ann['area'] <= 0 or w < 1 or h < 1:
                continue
`
area is apparently only computed from segmentation masks and not bounding boxes...thus if you have a custom dataset with no seg masks, your bboxes won't be loaded.  

I think just having the w and h check is sufficient and avoids people hitting this when training on custom datasets.
https://github.com/cocodataset/cocoapi/issues/36

**Comment by rwightman** [comment]
@lessw2020 good point, I'll make a note to fix that. BTW, if you haven't been following, the code on the `training` branch is working, sort of... I managed to find a few hours to adapt some training code from my image models repo for this task... 

So far it has
* distributed training with apex AMP + sync-bn support
* basic augmentations matching the TF automl code 
* ability to turn on EMA weight averaging

Major missing/broken things,
* need to release `timm` with a 'momentum' opt matching tf.train.MomentumOptimizer (SGD w/o Nesterov enabled by default)
* proper validation metric, currently just loss, no full coco eval or fast mAP calc being done
* saving the wrong part of model right now, saving full bench, need to save just .model
* full reproduction of automl level model accuracy, first crack only hit .26 mAP, this seems on par with other 3rd party attempts, but not good enough. It is most likely due to low LR caused by stability issues, I think I've fixed the stability issue and currently training at higher LR

**Comment by lessw2020** [comment]
Hi @rwightman,
Thanks for the update - that's great news as I'm full swing into setting up custom detector using EffDet and hoping to move from Signa over to your model soon :)

A couple points that may or may not be useful for you:
1 - I've been using augmentations from albumentations github with good success in terms of improving results (via loss) and they have a nice pipeline for chaining things together with OneOf and various level probabilities.
2 - Re: training and opt - I recently added gradient centralization to Ranger and saw a nice jump in obj detection training with it.  I'm also testing SPS right now and added in gradient centralization there and saw good success, so the net is you may want to try gc for improving results.  (not part of TF training but it's a nice improvement in most things I've tested it on.  )
3 - Proper loss metric would be a huge help.  Related, would you consider integrating Distance and/or complete IOU loss for the bbox regression?  https://github.com/Zzh-tju/DIoU
I haven't tested but it appears to be superiour than traditional regression setup. 
Thanks for the progress and updates and feel free to ping me if there are specific things I might be able to help out on.  
btw - do you envision this supporting ONNX or similar so it could be ported to TFLite for mobile there?  I've been working in TF directly and really not liking it but do need to be able to produce a mobile TFLite version for my work.  
Hope you are doing well!

**Comment by rwightman** [comment]
@lessw2020 extra goodies like DIoU etc will come in the experimentation phase, right now still trying to finish off the 'replicate official' baseline. I looked at albumentations but it seemed challenging to fit their defaults to replicate the Google baseline, also, their bbox format diff than the yxyx google, and they use lists of np array instead of one fat numpy array for some reason. 

For experiments I have an impl of EvoNorm on deck too, hopefully will make it possible to train the larger models without having access to clusters of V100s. Heh.

**Comment by lessw2020** [comment]
@rwightman  - great to see your progress on the training branch!  EvoNorm sounds great.  I'm going to get setup with your training branch tomorrow and start riding along and hopefully can help a bit.  
Yes albumentations has some oddities but the flip side is I will say it's very fast and has a lot of good augs though arguably most can be implemented pretty fast directly.  
Anyway, l greatly appreciate your progress on this and will get synced up tomorrow with the training branch. 
Hope you are doing well!
